client/helpers_test.go

```diff
@@ -968,3 +971,49 @@ func TestChangeTargetMetaFailsIfPrefixError(t *testing.T) {
 	require.Empty(t, repo.Targets[data.CanonicalTargetsRole].Signed.Targets)
 	require.Empty(t, repo.Targets["targets/level1"].Signed.Targets)
 }
+
+func TestAllNearExpiry(t *testing.T) {
+	repo, _, err := testutils.EmptyRepo("docker.com/notary")
+	require.NoError(t, err)
+	nearexpdate := time.Now().AddDate(0, 1, 0)
+	repo.Root.Signed.SignedCommon.Expires = nearexpdate
+	repo.Snapshot.Signed.SignedCommon.Expires = nearexpdate
+	repo.Targets["targets"].Signed.Expires = nearexpdate
+	_, err1 := repo.InitTargets("targets/exp")
+	require.NoError(t, err1)
+	repo.Targets["targets/exp"].Signed.Expires = nearexpdate
+	//Reset levels to display warnings through logrus
+	orgLevel := log.GetLevel()
+	log.SetLevel(log.WarnLevel)
+	defer log.SetLevel(orgLevel)
+	b := bytes.NewBuffer(nil)
+	log.SetOutput(b)
+	warnRolesNearExpiry(repo)
+	require.Contains(t, b.String(), "targets metadata is nearing expiry, you should re-sign the role metadata", "targets should show near expiry")
+	require.Contains(t, b.String(), "targets/exp metadata is nearing expiry, you should re-sign the role metadata", "targets/exp should show near expiry")
+	require.Contains(t, b.String(), "root is nearing expiry, you should re-sign the role metadata", "Root should show near expiry")
+	require.Contains(t, b.String(), "snapshot is nearing expiry, you should re-sign the role metadata", "Snapshot should show near expiry")
+}
```

Comment:
Timestamp isn't being warned about since @riyazdf mentioned that Notary deals with that role and the user needs to worry about it. Hence there is no warning output related to it. Also, the role will always throw a warning since defaults are lower than the 6 month threshold. Let me know what you think, I'm okay to do it either ways but what Riyaz mentioned seemed to make sense